includes/payment-tokens/class-wc-stripe-ach-payment-token.php

```diff
@@ -161,12 +161,10 @@ public function set_last4( $last4 ) {
 	 * @inheritDoc
 	 */
 	public function is_equal_payment_method( $payment_method ): bool {
-		if (
-			WC_Stripe_Payment_Methods::ACH === $payment_method->type
-			&& ( $payment_method->{WC_Stripe_Payment_Methods::ACH}->fingerprint ?? null ) === $this->get_fingerprint() ) {
-			return true;
+		if ( WC_Stripe_Payment_Methods::ACH !== $payment_method->type ) {
+			return false;
 		}
 
-		return false;
+		return ( $payment_method->{WC_Stripe_Payment_Methods::ACH}->fingerprint ?? null ) === $this->get_fingerprint();
```

Comment:
Might it be clearer to use the property name directly instead of the expression? (Definitely a question rather than a suggestion!)
```suggestion
		// ACH uses the `us_bank_account` property
		return ( $payment_method->us_bank_account->fingerprint ?? null ) === $this->get_fingerprint();
```

Reply:
I am OK with both versions. Applied the suggestions 👍